pkg/model/model.go

```diff
@@ -40,53 +41,36 @@ type Coeff struct {
 	InterceptDram float64 `csv:"intercept_dram"`
 }
 
+type EnergyPrediction struct {
+	Architecture   string
+	CPUTime        float64
+	CPUCycle       float64
+	CPUInstr       float64
+	ResidentMemory float64
+	CacheMisses    float64
+}
+
+type CategoricalFeature struct {
+	Name   string  `json:"name"`
+	Weight float64 `json:"weight"`
+}
+
+type NormalizedNumericalFeature struct {
+	Mean     float64 `json:"mean"`
+	Variance float64 `json:"variance"`
+	Weight   float64 `json:"weight"`
+}
+
 type CoreModelServerCoeff struct {
 	AllWeights struct {
 		BiasWeight           float64 `json:"Bias_Weight"`
 		CategoricalVariables struct {
-			CPUArchitecture struct {
-				AlderLake struct {
-					Weight float64 `json:"weight"`
-				} `json:"Alder Lake"`
-				Broadwell struct {
-					Weight float64 `json:"weight"`
-				} `json:"Broadwell"`
-				CascadeLake struct {
-					Weight float64 `json:"weight"`
-				} `json:"Cascade Lake"`
-				CoffeeLake struct {
-					Weight float64 `json:"weight"`
-				} `json:"Coffee Lake"`
-				Haswell struct {
-					Weight float64 `json:"weight"`
-				} `json:"Haswell"`
-				IvyBridge struct {
-					Weight float64 `json:"weight"`
-				} `json:"Ivy Bridge"`
-				SandyBridge struct {
-					Weight float64 `json:"weight"`
-				} `json:"Sandy Bridge"`
-				SkyLake struct {
-					Weight float64 `json:"weight"`
-				} `json:"Sky Lake"`
-			} `json:"cpu_architecture"`
+			CPUArchitecture []CategoricalFeature `json:"cpu_architecture"`
 		} `json:"Categorical_Variables"`
 		NumericalVariables struct {
-			CurrCPUCycles struct {
-				Mean     float64 `json:"mean"`
-				Variance int64   `json:"variance"`
-				Weight   float64 `json:"weight"`
-			} `json:"curr_cpu_cycles"`
-			CurrCPUTime struct {
-				Mean     float64 `json:"mean"`
-				Variance float64 `json:"variance"`
-				Weight   float64 `json:"weight"`
-			} `json:"curr_cpu_time"`
-			CurrentCPUInstructions struct {
-				Mean     float64 `json:"mean"`
-				Variance int64   `json:"variance"`
-				Weight   float64 `json:"weight"`
-			} `json:"current_cpu_instructions"`
+			CPUCycle NormalizedNumericalFeature `json:"curr_cpu_cycles"`
+			CPUTime  NormalizedNumericalFeature `json:"curr_cpu_time"`
+			CPUInstr NormalizedNumericalFeature `json:"current_cpu_instructions"`
```

Comment:
just a small comment, we should change json key to the same format and same name as up-to-date exported metrics (cpu_intruction --> cpu_instr, cache_misses  to cache_miss - this is for attacher package). Also, we may remove the `curr` or `current` prefix.